### PR TITLE
Add Nutanix as an external cloud provider platform

### PR DESCRIPTION
**Pull request by yanhua121**
Add Nutanix as an external cloud provider platform, in order to integrate the nutanix-ccm to openshift.

### CODE REVIEW

File: pkg/cloudprovider/external_test.go
Comment (JoelSpeed) [inline pkg/cloudprovider/external_test.go:176]: Please also add a test that shows what happens when there is a feature gate

**Review by JoelSpeed — COMMENTED** [review]
Once this is merged, you need to revendor the change in KCMO, KASO, MCO and CCMO. This will need to be co-ordinated. CCMO must come first, then KCMO, MCO and KASO in that order.

We must only do this once the image is in payload.

/hold until the Nutanix CCM image is in payload

File: pkg/cloudprovider/external_test.go
Comment (yanhua121) [reply pkg/cloudprovider/external_test.go:176]: Added the test cases.

**Comment by JoelSpeed** [comment]
Code changes LGTM, still waiting on the image to be in payload though
/lgtm

**Review by elmiko — COMMENTED** [review]
the image is now being included in the payload, i am removing the hold on this. @JoelSpeed ptal

/hold cancel
/lgtm

**Comment by JoelSpeed** [comment]
In terms of ordering for vendor updates after this merges, the important two to co-ordinate are CCMO and MCO. If you merge MCO first then you will have uninitialized machines and the cluster will not bootstrap, if you merge CCMO first there should be no ill-effect.

Merge CCMO, then MCO, then get the others updated please

**Comment by yanhua121** [comment]
> In terms of ordering for vendor updates after this merges, the important two to co-ordinate are CCMO and MCO. If you merge MCO first then you will have uninitialized machines and the cluster will not bootstrap, if you merge CCMO first there should be no ill-effect.
> 
> Merge CCMO, then MCO, then get the others updated please

@JoelSpeed Thanks for the guidance. Will follow it after this PR is merged.

**Review by soltysh — APPROVED** [review]
/approve

File: pkg/cloudprovider/external.go
Comment (elmiko) [inline pkg/cloudprovider/external.go:40]: typo here
```suggestion
		configv1.KubevirtPlatformType,
		configv1.NutanixPlatformType:
```

**Review by elmiko — COMMENTED** [review]
just a minor typo

**Comment by elmiko** [comment]
@soltysh tide doesn't seem to like the squash method

/label-remove tide/merge-method-squash

File: pkg/cloudprovider/external.go
Comment (yanhua121) [reply pkg/cloudprovider/external.go]: Fixed.

**Review by elmiko — COMMENTED** [review]
/lgtm

**Pull request merged**
